c7n/resources/dynamodb.py

```diff
@@ -450,3 +454,235 @@ class resource_type(object):
+                    'InvalidParameterValueException'):
+                log.warning(
+                    'Exception collecting tags for %s: \n%s' % (
+                        r['ClusterName'], e))
+        r['Tags'] = tags
+        return r
+
+    with executor_factory(max_workers=2) as w:
+        return list(w.map(process_tags, tables))
+
+
+@DynamoDbAccelerator.filter_registry.register('security-group')
+class DaxSecurityGroupFilter(SecurityGroupFilter):
+
+    RelatedIdsExpression = "SecurityGroups[].SecurityGroupIdentifier"
+
+
+@DynamoDbAccelerator.action_registry.register('tag')
+class DaxTagging(Tag):
+    """Action to create tag(s) on a resource
+
+        :example:
+
+        .. code-block:: yaml
+
+            policies:
+              - name: dax-cluster-tag
+                resource: dax
+                filters:
+                  - "tag:target-tag": absent
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-tag-value
+    """
+    permissions = ('dax:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('dax')
+        for r in resources:
+            try:
+                client.tag_resource(ResourceName=r[self.id_key], Tags=tags)
+            except ClientError as e:
+                if e.response['Error']['Code'] in (
+                        'ClusterNotFoundFault',
+                        'InvalidARNFault',
+                        'InvalidClusterStateFault',
+                        'InvalidParameterCombinationException',
+                        'InvalidParameterValueException'):
+                    self.log.warning(
+                        'Exception tagging %s: \n%s' % (
+                            r['ClusterName'], e))
+                    continue
+                raise
+
+
+@DynamoDbAccelerator.action_registry.register('remove-tag')
+class DaxRemoveTagging(RemoveTag):
+    """Action to remove tag(s) on a resource
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: dax-remove-tag
+            resource: dax
+            filters:
+              - "tag:OutdatedTag": present
+            actions:
+              - type: remove-tag
+                tags: ["OutdatedTag"]
+    """
+    permissions = ('dax:UntagResource',)
+
+    def process_resource_set(self, resources, tag_keys):
+        client = local_session(
+            self.manager.session_factory).client('dax')
+        for r in resources:
+            try:
+                client.untag_resource(
+                    ResourceName=r['ClusterArn'], TagKeys=tag_keys)
+            except ClientError as e:
+                if e.response['Error']['Code'] in (
+                        'ClusterNotFoundFault',
+                        'InvalidARNFault',
+                        'InvalidClusterStateFault',
+                        'InvalidParameterCombinationException',
```

Comment:
ditto for invalidparameter* errors as their persistent errors and should raise, nice find on tagnotfoundfault